torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -160,7 +160,10 @@ def inner(*flat_args):
         with disable_above, mode:
             # precondition: The passed in function already handles unflattening inputs + flattening outputs
             flat_f_args = pytree.tree_map(_to_fun, flat_args)
+
+            tmp_counter = torch.nested._internal.nested_tensor._tensor_id_counter
             flat_f_outs = f(*flat_f_args)
+            torch.nested._internal.nested_tensor._tensor_id_counter = tmp_counter
```

Comment:
Maybe there's a more principled way to do this

Reply:
Without this change, the case where NJT is created from offsets which itself is an intermediate tensor would break.

It should be breaking because (1) aot autograd traces forward twice, one to record some aliasing metadata, and a second time to trace out the joint graph, (2) aot autograd creates tangents from the output of the first forward to use as inputs to the joint graph.

Today when we trace, we mutate the global state, meaning that if we try to trace multiple times we produces different results, leading to a mismatch in shapes.

(Also what we may want to do in addition to resetting is to having a separate counter during tracing, so that the eager state is not affected.)

> maybe a context manager?

Yeah, this may be better if we want to stick with this solution, but I was lazy here lol